symplyphysics/laws/thermodynamics/equations_of_state/van_der_waals/dimensionless_equation.py

```diff
@@ -30,7 +33,33 @@
     Rational(8, 3) * reduced_temperature,
 )
 
-# TODO: derive from van der Waals equation of state
+# Derive from van der Waals equation of state
+
+_critical_pressure_expr = critical_pressure.law.rhs.subs({
+    critical_pressure.bonding_forces_parameter: equation.bonding_forces_parameter,
+    critical_pressure.molecules_volume_parameter: equation.molecules_volume_parameter,
+})
+
+_critical_molar_volume_expr = critical_molar_volume.law.rhs.subs({
+    critical_molar_volume.molecules_volume_parameter: equation.molecules_volume_parameter,
+})
+
+_critical_temperature_expr = critical_temperature.law.rhs.subs({
+    critical_temperature.bonding_forces_parameter: equation.bonding_forces_parameter,
+    critical_temperature.molecules_volume_parameter: equation.molecules_volume_parameter,
+})
+
+_reduced_eqn = equation.law.subs({
+    equation.pressure: reduced_pressure * _critical_pressure_expr,
+    equation.volume: reduced_volume * _critical_molar_volume_expr * equation.amount_of_substance,
```

Comment:
It seems, we have molar heat capacity definition, but no other `molar` parameters. Can we write some general law of `molar` something and use it here for calculating critical volume?

Reply:
You mean like a general law that relates a non-molar and a molar quantity, say, $`X = X_m \cdot n`$, with $`X`$ and $`X_m`$ being `sympy.Symbol`s?

Reply:
Yeah